website/docs/components/code-editor/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,106 @@
+## Usage
+
+### When to use
+
+- When a user needs to create code from scratch.
+- When code requires modification from an existing source.
+
+### When not to use
+
+- As a read-only reference for code, use the [Code Block](/components/code-block) instead.
+
+## Standalone
+
+The `isStandalone` property increases the portability of the Code Editor to ensure that it can be used in different contexts. For example, a common use case of the Code Editor is in a "standalone" context, which can be part of a form or multi-step process and is generally a part of the normal layout flow.
+
+![Code Editor with rounded corners for enabled standalone property](/assets/components/code-editor/code-editor-rounded-standalone.png)
+
+Sometimes it may be necessary to use the Code Editor in a more dense layout or nested within another component.
+
+![Code Editor embedded inside of a logs UI, where the corners are squared to fit within container.](/assets/components/code-editor/code-editor-block-level.png)
+
+## Header
+
+An optional header is available within the Code Editor. It consists of three sections that can be toggled as needed:
+
+- the text content that includes the title and description
+- an actions container for secondary actions
+- a custom content section for adding primary actions
+
+### When to use a title and description
+
+A title and description provides additional contextual information for the Code Editor. An accessible name is mandatory and can either be provided by the title or by an external text element.
+
+!!! Do
+
+Provide an accessible name for the Code Editor so that users of assistive technology can understand its purpose.
+
+![A Code Editor embedded in a form following a set of radio buttons. It has the heading "Policy code (Sentinel)" immediately before.](/assets/components/code-editor/code-editor-external-do-accessible-name.png)
+
+!!!
+
+!!! Dont
+
+Assume all users will understand the purpose of the Code Editor without providing an accessible name. Not providing one may cause confusion and is an accessibility failure.
+![A Code Editor embedded in a form following a set of radio buttons. The Code Editor has no title labelling it.](/assets/components/code-editor/code-editor-dont-external-accessible-name.png)
+
+!!!
+
+## Secondary actions
+
+The secondary actions section supports two optional buttons: the [Copy Button](/components/copy/button), and a “full screen” button. The Copy Button copies the content of the Code Editor to the clipboard, while the full screen button toggles the size of the Code Editor from inline view to full screen.
```

Comment:
Comment: We should try to follow a consistent style for quotes, at least within the same “page”.

In this text, curly quotes are used:
... and a “full screen” button...

In other text, straight quotes are used:
... heading "Policy code (Sentinel)" immediately ...

(My personal preference would be to use typographic quotes, i.e. curly quotes but we haven‘t really been following that.)